packages/codemod/src/utils/prop-utils.ts

```diff
@@ -1,31 +1,129 @@
+export function isPropExists(j: JSCodeshift, elementPath: ASTPath<JSXElement>, propName: string): boolean {
+  return j(elementPath).find(JSXOpeningElement).find(JSXIdentifier, { name: propName }).size() > 0;
+}
+
+/**
+ * Finds props on a JSX element
+ */
+export function findProps(j: JSCodeshift, elementPath: ASTPath<JSXElement>, ...propNames: string[]) {
+  return j(elementPath)
+    .find(JSXOpeningElement)
+    .find(JSXAttribute)
+    .filter(attr => {
+      const attrName = attr.node.name.type === "JSXIdentifier" ? attr.node.name.name : "";
+      return propNames.includes(attrName);
+    });
+}
+
+/**
+ * Removes a prop from a JSX element
+ */
+export function removeProp(j: JSCodeshift, elementPath: ASTPath<JSXElement>, ...propsNames: string[]): void {
+  j(elementPath)
+    .find(JSXOpeningElement)
+    .find(JSXAttribute)
+    .forEach(attrPath => {
+      if (attrPath.node.name.type === "JSXIdentifier" && propsNames.includes(attrPath.node.name.name)) {
+        j(attrPath).remove();
+      }
+    });
+}
+
+/**
+ * Gets the computed value of a prop
+ */
+function getPropValue(j: JSCodeshift, prop: JSXAttribute): undefined | boolean | Literal["value"] | string {
+  const value = prop.value;
+
+  // boolean flag value (e.g. <Button disabled />)
+  if (value === null) {
+    return true;
+  }
+
+  // string value (e.g. <Button text="Click me" />)
+  if (value?.type === "Literal") {
+    return value.value;
+  }
+
+  // expression value (e.g. <Button text={...} />).
+  if (value?.type === "JSXExpressionContainer") {
+    if (value.expression.type === "Literal") {
+      // e.g. <Button text={true} /> or <Button text={"text"} />
+      return value.expression.value;
+    }
+  }
+
+  // can be very complex, we'll have to compare strings
+  return value ? j(value).toSource() : undefined;
+}
+
+/**
+ * Checks if a Collection of props have the same value
+ */
+export function propsValueMatch(j: JSCodeshift, propsCollection: Collection<JSXAttribute>): boolean {
+  const values = propsCollection.nodes().map(prop => getPropValue(j, prop));
+  return values.every(value => value === values[0]);
+}
+
+/**
+ * Perform a regular prop migration on a JSX element, and logs an error if conditions are not met
+ * 1. If new prop is not found, update the prop name
+ * 2. If both props are found, check if the values match
+ *   - If they do, remove the old prop
+ *   - If they don't, log an error
+ */
+export function migratePropsNames(
+  j: JSCodeshift,
+  elementPath: ASTPath<JSXElement>,
+  filePath: string,
+  componentName: string,
+  propsNamesMappingOldToNew: Record<string, string>
+): void {
+  Object.entries(propsNamesMappingOldToNew).forEach(([deprecatedPropName, newPropName]) => {
+    const props = findProps(j, elementPath, deprecatedPropName, newPropName);
+
+    if (props.length !== 2) {
+      updatePropName(j, elementPath, { [deprecatedPropName]: newPropName });
+      return;
+    }
+
+    if (propsValueMatch(j, props)) {
```

Comment:
Not needed for now, but let's remember for the future that this will work only on strings and booleans, but won't catch arrays and objects

Reply:
it actually does catch arrays and objects, but in a very high level way
e.g. this would work:
```
const someArray = [1, 2, 3];
<MyComponent myArray={someArray} myNewArray={someArray} />
```

but for now, this won't work
```
const someArray = [1, 2, 3];
<MyComponent myArray={someArray} myNewArray={[1, 2, 3]} />
```
and honestly I'm not sure this **should** work as this can have different meaning in code, stuff that are done on purpose

so right now we're catching
<MyComponent **someFlag** **someFlag={true}** **someFlag={false}** **someString={"hello"}** someString="hello" />
which I think are more than enough for now

Reply:
Yes, this is what I meant, if it is the same value but different pointer. 
I think this is fine for now, but some of our props' types are array so we need to keep that in mind for later

Reply:
yes so for now it will check exact match
also cases like
```onClick={() => doSomething()} and onClick={doSomething}```
won't be caught as equals, but that is also cool I think, until we'll encounter somewhere where a specific use case is needed